packages/dds/sequence-sequential-id/src/sharedStringWithSequentialId.ts

```diff
@@ -0,0 +1,88 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import { Lazy } from "@fluidframework/common-utils";
+import { IChannelAttributes, IFluidDataStoreRuntime } from "@fluidframework/datastore-definitions";
+import { Marker, MergeTreeDeltaType, MergeTreeMaintenanceType,
+     reservedMarkerIdKey, SortedSegmentSet } from "@fluidframework/merge-tree";
+import { SharedString, SequenceDeltaEvent, SequenceMaintenanceEvent } from "@fluidframework/sequence";
+import { createIdBetween, digitLast, digitZero } from "./generateSequentialId";
+import { SharedStringWithSequentialIdFactory } from "./SharedStringWithSequentialIdFactory";
+
+export function sharedStringWithSequentialIdMixin(Base: typeof SharedString = SharedString): typeof SharedString {
+    return class SharedStringWithSequentialId extends Base {
+        private readonly sortedMarkers: Lazy<SortedSegmentSet>;
+        constructor(document: IFluidDataStoreRuntime, public id: string, attributes: IChannelAttributes) {
+            super(document, id, attributes);
+            this.sortedMarkers = new Lazy<SortedSegmentSet>(() => {
+                const set = new SortedSegmentSet();
+                this.walkSegments((segment) => {
+                    if (Marker.is(segment) && segment.getId() !== undefined) {
+                        // If it has an id then we have acked it
+                        set.addOrUpdate(segment);
+                    }
+                    return true;
+                });
+                return set;
+            });
+            this.on("maintenance", this.applyIdToLocalAckedSegment);
+            this.on("sequenceDelta", this.applyIdToRemoteAckedSegment);
```

Comment:
Do we need to be on both events? do we get acks (for local and remote) through "maintenance" event?

Reply:
Yes... I chatted with Tony, we need to listen to both events. We will only get acks for local changes, so we need the sequenceDelta one to listen to remote events. Let me add comments for this

Reply:
Added comments